heartbeat/docs/autodiscover-hints.asciidoc

```diff
@@ -0,0 +1,40 @@
+{beatname_uc} supports autodiscover based on hints from the both Docker and Kubernetes. The hints system looks for
+hints in Kubernetes Pod annotations or Docker labels that have the prefix `co.elastic.monitor`. As soon as
+the container starts, {beatname_uc} will check if it contains any hints and launch the proper config for
+it. Hints tell {beatname_uc} how to get logs for the given container.
+
+Hints for Heartbeat take the form `co.elastic.monitor/MONITOR_INDEX.CONFIG_OPTION=VALUE`. Where `MONITOR_INDEX` represents
+the index of monitor, `CONFIG_OPTION` is any of the <<configuration-heartbeat-options>> options, with `VALUE` being the value
+you'd like to set that option to.
+
+As an example let's convert the regular syntax heartbeat monitors below into autodiscover hints.
+
+```yaml
```

Comment:
Use asciidoctor source code block formatting with syntax highlighting here (what you have here is Markdown). See https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#source-code